bin/exec.ml

```diff
@@ -93,18 +93,36 @@ let term =
         let prog = Path.extend_basename prog ~suffix:Bin.exe in
         Option.some_if (Path.exists prog) prog
   in
+  (* Good candidates for the "./x.exe" instead of "x.exe" error are executables
+     present at the root of the build dir *)
```

Comment:
The command is interpreted relative to the current directory. More precisely, the two commands will run the same program:

- `dune exec foo/bar.exe`
- `cd foo && dune exec ./bar.exe`

so rather than the targets at the root of the build dir, we should look for the targets in the current directory. You can use `Build_system.targets_of ~dir` for that.